pkg/webhook/tenant_prefix/patching.go

```diff
@@ -49,18 +51,26 @@ func (o Webhook) GetPath() string {
 }
 
 type handler struct {
-	forceTenantPrefix bool
+	forceTenantPrefix        bool
+	protectedNamespacesRegex *regexp.Regexp
 }
 
 func (r *handler) OnCreate(ctx context.Context, req admission.Request, clt client.Client, decoder *admission.Decoder) admission.Response {
-	if !r.forceTenantPrefix {
-		return admission.Allowed("")
-	}
 	ns := &corev1.Namespace{}
 	if err := decoder.Decode(req, ns); err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
+	if r.protectedNamespacesRegex != nil {
+		if matched := r.protectedNamespacesRegex.MatchString(ns.GetName()); matched {
+			return admission.Denied("Creating namespaces with name matching " + r.protectedNamespacesRegex.String() + " regexp is not alloweded; please, reach out the system administrators")
```

Comment:
Should the error be `Creating namespace with name non-matching `r.protectedNamespacesRegex.String()` regex is not allowed`?

Take a look also to the typo on `alloweded`

Reply:
fixed